File: tareas-entregables/primera/sinesum1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sinesum1
{
    class Program
    {
        static void GetEvenlyDistPoints(int numPoins, double min, double max, ref double[] lista)
        {
            lista[0] = min;
            double increment = (max - min) / (numPoins - 1);
            for (int i = 1; i < numPoins - 1; i++)
            {
                lista[i] = lista[i - 1] + increment;
            }
            lista[lista.Length - 1] = max;
        }
        static double[] S(ref double[] time,int n_iter, double T)
        {
            double[] aux_list = new double[time.Length];
            double aux_acc = 0;
            
            for(int j=0; j<time.Length; j++)
            {
                for (int i = 1;i <= n_iter; i++)
                {//Recordar siempre poner los numeros como dobles ej 2.0
                    aux_acc = aux_acc + (1.0 / ((2 * i) - 1)) * Math.Sin((2.0 * (2 * i - 1) * Math.PI * time[j]) / T);
                }
                aux_list[j] = (4 / Math.PI) * aux_acc ;
                aux_acc = 0;
            }
            return aux_list;
        }
        static double[] f(ref double[] t,double T) {
            double[] aux_list = new double[t.Length];
            for (int i = 0; i< t.Length;i++) 
            {
                if (t[i] > 0 && t[i] < T / 2) 
                {
                    aux_list[i] = 1;
                }
                else if (t[i] == T / 2) 
                {
                    aux_list[i] = 0;
                }
                else if (t[i] > T / 2 && t[i]<T) 
                {
                    aux_list[i] = -1;
                }
                else 
                {
                    Console.WriteLine("t out of range, error on the arguments!!");
                }
            }
            return aux_list;
        }
        static void Main(string[] args)
        {
            int n = 5;
            double T = 2 * Math.PI;
            double[] alpha = new double[50];
            GetEvenlyDistPoints(alpha.Length, 0.01, 0.99, ref alpha);
      
            double[] t = new double[alpha.Length];
            
            for (int i = 0; i< t.Length; i++)
            {
                t[i] = T * alpha[i];
            }
       
            double[] St = S(ref t, n, T);
            double[] ft = f(ref t, T);

            Console.WriteLine("N# |f(t)|S(t)|err|");
            Console.WriteLine("-------------------------");
            for(int i = 0; i < St.Length; i++)
            {
                Console.WriteLine("{0} | {1} | {2} | {3} ",i,ft[i],St[i],ft[i]-St[i]);
            }
        }
    }
}

File: tareas-entregables/primera/sinesum1.py
import math
import matplotlib.pyplot as plt
import numpy as np

def S(time,n_iter,T):
    aux_list=[]
    aux_acc= 0
    for t in time:
        for i in n_iter:
            aux_acc += (1/(2*i - 1))*math.sin((2*(2*i-1)*math.pi*t)/T)
            print(aux_acc)
        aux_list.append((4/math.pi)*aux_acc)
        aux_acc = 0
    return aux_list    

def f(t,T):
    aux_list = []
    for p in t:
        if p>0 and p<T/2:
            aux_list.append(1)
        elif p==T/2:
            aux_list.append(0)
        elif p>T/2 and p<T:
            aux_list.append(-1)
        else:
            print('t out of range, error on the arguments!!')
    return aux_list

if __name__ == "__main__":
    top=5
    n = range(1,top+1)
    T=2*math.pi
    alpha = np.linspace(0.01,0.99,50)
    t = [ T*e for e in alpha ]
    print(t)
    St = S(t,n,T)
    ft = f(t,T)

    plt.plot(t,St,'g--')
    plt.plot(t,ft,'b.-')
    plt.show()
    print(f'N# |f(t)|S(t)|err|')
    print(f'-------------------------')
    for i in range(0,len(St)):
        print(f'{i}  | {ft[i]} | {St[i]:.3f} | {(ft[i]-St[i]):.3f}')